src/khoj/routers/api_chat.py

```diff
@@ -810,38 +829,52 @@ async def send_rate_limit_message(message: str):
 
                     for webpage in direct_web_pages[query]["webpages"]:
                         webpages.append(webpage["link"])
-
-                await send_status_update(f"**📚 Read web pages**: {webpages}")
+                async for result in send_event("status", f"**📚 Read web pages**: {webpages}"):
+                    yield result
             except ValueError as e:
                 logger.warning(
-                    f"Error directly reading webpages: {e}. Attempting to respond without online results", exc_info=True
+                    f"Error directly reading webpages: {e}. Attempting to respond without online results",
+                    exc_info=True,
                 )
 
+        ## Send Gathered References
+        async for result in send_event(
+            "references",
```

Comment:
What do you think of defining an Enum for the well-known event types we're sending to the client? It'll help organize/wrangle the information passage here a bit better.

Reply:
Good idea! Updated code to use enum to track chat stream event types